### DISCUSSION TITLE
Philips Hue Smart Plug: missing state update on button press

### DISCUSSION THREAD

**Discussion by michel-zedler** [Koenkk/zigbee2mqtt#10912] [Devices]
I have several identical (?) Hue Smart Plug plugs, all paired successfully, can be switched via MQTT as documented here https://www.zigbee2mqtt.io/devices/929003050601.html
and pressing the hardware button will toggle the state (ON/OFF) as expected.

But one plug (0x...4f) consistently publishes a state update in MQTT (ON/OFF) after the button press - while the others don't.

Can you guide me how to troubleshoot this?

Device:
Manufacturer: Philips
Name: Hue Smart Plug
Model: ‎929003050601 (as ordered)
Model: 9290030506 (as printed on the plug) 

Using Zigbee2MQTT 1.22.1
Adapter hardware: CC2652RB from https://slae.sh/projects/cc2652/

All plugs were identified as: Philips Hue smart plug (929003050601)

```
Zigbee2MQTT:info  2022-01-11 10:53:58: Device '0x...4f' is supported, identified as: Philips Hue smart plug (929003050601)
Zigbee2MQTT:info  2022-01-11 17:13:50: Device '0x...ef' is supported, identified as: Philips Hue smart plug (929003050601)
...
```

Same behaviour after device firmware updates:

```
Zigbee2MQTT:info  2022-01-19 11:42:07: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"id":"0x...4f"},"meta":{"device":"0x...4f","from":{"dateCode":"20201117","softwareBuildID":"1.76.11"},"status":"update_succeeded","to":{"dateCode":"20211119","softwareBuildID":"1.93.6"}},"type":"ota_update"}'
Zigbee2MQTT:info  2022-01-20 09:17:46: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"id":"0x...ef"},"meta":{"device":"0x...ef","from":{"dateCode":"20201117","softwareBuildID":"1.76.11"},"status":"update_succeeded","to":{"dateCode":"20211119","softwareBuildID":"1.93.6"}},"type":"ota_update"}'
...
```

Same behaviour after upgrading from Zigbee2MQTT version: 1.22.1 to 1.22.2

**Comment by michel-zedler** [comment]
If you have access to a Philips Hue Smart Plug: what is your experience, does it publish state changes triggered by the hardware button on the plug?

**Reply by Koenkk** [reply]
Could you provide the debug log when clicking the yellow reconfigure button in the z2m frontend?

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable debug logging.

**Comment by michel-zedler** [comment]
Hi @Koenkk ,

I reconfigured all affected devices, logs all same like:
```
debug 2022-01-27 12:21:12: Received MQTT message on 'zigbee2mqtt/bridge/request/device/configure' with data '{"id":"0x00...","transaction":"ujzuj-2"}'
info  2022-01-27 12:21:12: Configuring '0x00...'
info  2022-01-27 12:21:12: Successfully configured '0x00...'
info  2022-01-27 12:21:12: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/configure', payload '{"data":{"id":"0x00..."},"status":"ok","transaction":"ujzuj-2"}'
```
and I still got the same behaviour.

Meanwhile I have removed two SONOFF smart plugs from my network as they had their own problems, see  https://github.com/Koenkk/zigbee2mqtt/issues/10282#issuecomment-1017388555.
I have also unplugged and plugged all Hue smart plugs.

Next time I checked **all Hue Smart Plugs have started reporting state updates on device button press:**

```
debug 2022-01-27 12:07:16: Device '0x...ef' announced itself
info  2022-01-27 12:07:16: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x...ef","ieee_address":"0x...ef"},"type":"device_announce"}'
info  2022-01-27 12:07:16: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x...ef"},"type":"device_announced"}'
debug 2022-01-27 12:07:18: Retrieving state of '0x...ef' after reconnect
debug 2022-01-27 12:07:18: Received Zigbee message from '0x...ef', type 'readResponse', cluster 'genOnOff', data '{"onOff":1}' from endpoint 11 with groupID 0
info  2022-01-27 12:07:18: MQTT publish: topic 'zigbee2mqtt/0x...ef', payload '{"linkquality":90,"state":"ON","update":{"state":"idle"},"update_available":false}'
debug 2022-01-27 12:07:24: Received Zigbee message from '0x...ef', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 11 with groupID 0
info  2022-01-27 12:07:24: MQTT publish: topic 'zigbee2mqtt/0x...ef', payload '{"linkquality":96,"state":"ON","update":{"state":"idle"},"update_available":false}'

<!-- Hue smart plug BUTTON press --> 

debug 2022-01-27 12:08:39: Received Zigbee message from '0x...ef', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 11 with groupID 0
info  2022-01-27 12:08:39: MQTT publish: topic 'zigbee2mqtt/0x...ef', payload '{"linkquality":90,"state":"OFF","update":{"state":"idle"},"update_available":false}'

<!-- Hue smart plug BUTTON press --> 

debug 2022-01-27 12:09:01: Received Zigbee message from '0x...ef', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 11 with groupID 0
info  2022-01-27 12:09:01: MQTT publish: topic 'zigbee2mqtt/0x...ef', payload '{"linkquality":90,"state":"ON","update":{"state":"idle"},"update_available":false}'
```

So I think the failing SONOFF coordinators might have caused the problem.
I'm just not 100% sure because I was always able to externally switch all Hue smart plugs via MQTT.

**Comment by synio-wesley** [comment]
I actually have the same problem with my Philips Hue Smart Plug. But I don't have any SONOFF coordinators. I do relay ZGP messages from a kinetic switch through the Philips Hue Smart Plug though. But I can't imagine that to be a problem? (I don't have a Philips Hue Bridge unfortunately, so I can't update the firmware of the Smart Plug.)

Edit: Nevermind, I was able to fix it! Apparently, I could actually perform an OTA update with my dongle by sending to `zigbee2mqtt/bridge/request/device/ota_update/update` and after that, it started working!
